### PR TITLE
Heartbeats during snapshot install

### PR DESCRIPTION
**Pull request by MathieuBordere**
This would resolve https://github.com/canonical/raft/issues/179 . Currently SnapshotInstall RPCs are abused as a sort of heartbeat during follower snapshot installation, but this is unwanted due to the potentially very large size of the SnapshotInstall RPC.

### CODE REVIEW

File: include/raft.h [hunk 1]
```diff
@@ -508,12 +508,13 @@ enum { RAFT_UNAVAILABLE, RAFT_FOLLOWER, RAFT_CANDIDATE, RAFT_LEADER };
  */
 struct raft_progress
```
Comment (stgraber) [inline include/raft.h:509]: Just confirming (sorry, not familiar with code), but this struct is in-memory only and never persisted to disk or exchange over the network, right?

Always triggered by addition of struct members in the middle of a struct ;)

Comment (MathieuBordere) [reply include/raft.h:509]: Yes, it's in-memory only.

**Comment by MathieuBordere** [comment]
@freeekanayaka Any feedback on this?

**Comment by freeekanayaka** [comment]
Seems a fine approach to me.

**Pull request merged**